File: Chain of Responsibility/PHP/Caching.php
<?php
require_once('Request.php');
require_once('Handler.php');

class Caching extends Handler {
    private $data = [];

    function execute(Request $request) {
        if (array_key_exists($request->get_route(), $this->data)) {
            echo "Cached\n";

            return $this->data[$request->get_route()];
        } else {
            echo "Uncached\n";

            $temp_data = $this->next_handler->execute($request);
            $this->data[$request->get_route()] = $temp_data;

            return $temp_data;
        }
    }
}

File: Chain of Responsibility/Python 3/Caching.py
from Handler import Handler
from Response import Response

class Caching(Handler):
  def __init__(self) -> None:
    super().__init__()
    self.data = {}

  def execute(self, request):
    if request.route in self.data:
      print("Cached")

      return self.data[request.route]
    else:
      temp_data = self.next_handler.execute(request)
      self.data[request.route] = temp_data

      return temp_data
